### PR TITLE
Fixed display of energy label from plugin wc-eu-energy-label could not be disabled by default.

### PR DESCRIPTION
**Pull request by juanlopez4691**
### Ticket
- [Add new checkbox in order to disable energy label displaying per default](https://app.asana.com/0/1156709897787098/1151258293585977)

### Description
-

### CODE REVIEW

File: src/WooCommerce.php [hunk 1]
```diff
@@ -270,6 +289,34 @@ public static function woocommerce_product_options_general_product_data() {
...
+   *
+   * The backend seting for shop-standards overrides the product setting
+   * toggling the display of the energy label added by plugin
+   * wc-eu-energy-label.
+   *
+   * @implements get_post_metadata
+   */
+  public static function get_post_metadata($value, $object_id, $meta_key, $single) {
+    if ($meta_key === 'wc_eu_energy_label_disable_frontend') {
+      global $wpdb;
```
Comment (fabianmarz) [inline src/WooCommerce.php:303]: Globals should be defined at the very top of a function body.

File: src/WooCommerce.php [hunk 2]
```diff
@@ -270,6 +289,34 @@ public static function woocommerce_product_options_general_product_data() {
...
+
+      $disableLabel = wc_string_to_bool($wpdb->get_var(
```
Comment (fabianmarz) [inline src/WooCommerce.php:305]:
```suggestion
      $disable_label = wc_string_to_bool($wpdb->get_var(
```

File: src/WooCommerce.php [hunk 3]
```diff
@@ -270,6 +289,34 @@ public static function woocommerce_product_options_general_product_data() {
...
+        $wpdb->prepare(
+          "SELECT meta_value FROM $wpdb->postmeta
+          WHERE post_id = %s
+          AND meta_key = %s",
+          $object_id,
+          'wc_eu_energy_label_disable_frontend'
+        )
+      ));
+      $forceDisableLabel = wc_string_to_bool(get_option('_' . Plugin::L10N . '_disable_energy_label'));
```
Comment (fabianmarz) [inline src/WooCommerce.php:314]:
```suggestion
      $force_disable_label = wc_string_to_bool(get_option('_' . Plugin::L10N . '_disable_energy_label'));
```

File: src/WooCommerce.php [hunk 4]
```diff
@@ -270,6 +289,34 @@ public static function woocommerce_product_options_general_product_data() {
...
+      $value = $disableLabel || $forceDisableLabel ? 'yes' : 'no';
```
Comment (fabianmarz) [inline src/WooCommerce.php:315]:
```suggestion
      $value = $disable_label || $force_disable_label ? 'yes' : 'no';
```

**Comment by juanlopez4691** [comment]
This PR is discarded. After discussing with @Stephan Sperling we decided this feature would be better implemented in the new energy label plugin @Luca Pipolo (away through Friday) is working on. https://github.com/netzstrategen/woocommerce-energy-label

**Comment by juanlopez4691** [comment]
This PR is discarded. After discussing with @Stephan Sperling we decided this feature would be better implemented in the new energy label plugin @luca Pipolo (away through Friday) is working on. https://github.com/netzstrategen/woocommerce-energy-label

**Pull request closed without merging**